### PR TITLE
simplejson: create errors.pyi

### PR DESCRIPTION
**Pull request by Avasam**
This type stub file was generated by pyright. Then filled in manually.

### CODE REVIEW

**Comment by Avasam** [comment]
Is Python 2.7 and 3.6 support necessary?

File: stubs/simplejson/simplejson/errors.pyi [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    msg: str = ...
+    doc: str = ...
+    pos: int = ...
+    end: int | None = ...
+    lineno: int = ...
+    colno: int = ...
+    endlineno: int | None = ...
+    endcolno: int | None = ...
+    def __init__(self, msg: str, doc: str, pos: int, end: int | None = ...) -> None: ...
+    def __reduce__(self) -> tuple[type[JSONDecodeError], tuple[str, str, int, int]]: ...
```
Comment (Akuli) [inline stubs/simplejson/simplejson/errors.pyi:34]:
```suggestion
    def __reduce__(self) -> tuple[type[JSONDecodeError], tuple[str, str, int, int | None]]: ...
```

File: stubs/simplejson/simplejson/errors.pyi [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    """
+
+    msg: str = ...
+    doc: str = ...
+    pos: int = ...
+    end: int | None = ...
+    lineno: int = ...
+    colno: int = ...
+    endlineno: int | None = ...
+    endcolno: int | None = ...
```
Comment (Akuli) [inline stubs/simplejson/simplejson/errors.pyi:32]: The `= ...` here does nothing for attributes, and we prefer leaving it out.

```suggestion
    msg: str
    doc: str
    pos: int
    end: int | None
    lineno: int
    colno: int
    endlineno: int | None
    endcolno: int | None
```

File: stubs/simplejson/simplejson/errors.pyi [hunk 3]
```diff
@@ -0,0 +1,34 @@
+"""Error classes used by simplejson
```
Comment (Akuli) [inline stubs/simplejson/simplejson/errors.pyi:1]: Please delete both docstrings. See #5444 for discussion.

**Comment by Akuli** [comment]
>Is Python 2.7 and 3.6 support necessary?

Currently the simplejson stubs support Python 2.7, because there is `python2 = true` in METADATA.toml. If you need something that doesn't work in Python 3.7, we can remove the `python2 = true`. We also support Python 3.6.

However, this in fact has nothing to do with `from __future__ import annotations`. In stub files, `from __future__ import annotations` is always done implicitly, and you shouldn't do it yourself at all, regardless of which Python version the stubs are targeting.

**Comment by Avasam** [comment]
> In stub files, `from __future__ import annotations` is always done implicitly, and you shouldn't do it yourself at all, regardless of which Python version the stubs are targeting.

Thanks I did not know that. I have Flake8 installed and it was whining about the self-reference.

**Comment by Akuli** [comment]
flake8 needs a plugin to work correctly with pyi files: https://github.com/python/typeshed/blob/bb7a06c60b0a7187c46729fdc9207b1e139e3cf9/requirements-tests-py3.txt#L7

Nit: Avoid force-pushing, because it makes it harder to see what changed since the previous review. When merging the pull request, your commits will be combined into a single commit with GitHub's "Squash and merge" button anyway.

**Review by Akuli — APPROVED** [review]

**Comment by Akuli** [comment]
Thanks!

**Comment by Avasam** [comment]
> flake8 needs a plugin to work correctly with pyi files:
> 
> https://github.com/python/typeshed/blob/bb7a06c60b0a7187c46729fdc9207b1e139e3cf9/requirements-tests-py3.txt#L7
> 
> Nit: Avoid force-pushing, because it makes it harder to see what changed since the previous review. When merging the pull request, your commits will be combined into a single commit with GitHub's "Squash and merge" button anyway.

It's a habit taken to keep the commit history from pull requests clean. But I'll try to keep in mind thank you!

**Pull request merged**